src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -812,8 +820,9 @@ private void updateEditState() {
             {
                 for (Pair<String, JSONObject> changeInfo:mEditorChanges.values()) {
                     try {
-                        final ViewVisitor visitor = mProtocol.readEdit(changeInfo.second);
-                        newEdits.add(new Pair<String, ViewVisitor>(changeInfo.first, visitor));
+                        final EditProtocol.Edit edit = mProtocol.readEdit(changeInfo.second);
+                        newEdits.add(new Pair<String, ViewVisitor>(changeInfo.first, edit.visitor));
```

Comment:
I can understand what this logic does but it looks a bit confusing to me.. We have variables called `newEdits` and `edit`. However, when we add new elements to newEdits, we are not adding edit, but `new Pair<String, ViewVisitor>(changeInfo.first, edit.visitor)` (`edit.visitor` but not `edit`.. and ideally, it should be `newEdits.add(edit)`). Not a big deal, but the logic is different from what the variable names imply. Never mind, any change to newEdits will affect a bunch of stuff. It's ok to leave as it is.